### PR TITLE
[@mantine/core] Progress: Add several props

### PR DESCRIPTION
**Pull request by Ultra-Instinct-05**
This PR proposes to add the following props to `Progress`.

1. Adds the `isIndeterminate` prop for striped, animated progress bars with `value = 100`. If this is set, the `value` prop is ignored. It's `false` by default.
2. Adds the `animated` prop for animating striped progress bars. It's `false` by default.
3. Adds the `label` prop which is a `string` that can be used to set any text within the progress bar. Useful if you want to indicate the current value in the progress bar (or any other form of metric related to `value`)
4. Adds the `showLabel` prop that controls whether the `label` get's shown. It defaults to `false`.

The tests seem to be failing but it seems to be related to #562

### CODE REVIEW

File: src/mantine-core/src/components/Progress/Progress.test.tsx [hunk 1]
```diff
@@ -23,7 +23,12 @@ describe('@mantine/core/Progress', () => {
   itSupportsSx(Progress, defaultProps);
   itSupportsRef(Progress, defaultProps, HTMLDivElement);
   checkAccessibility([mount(<Progress value={80} aria-label="test-progress" />)]);
-  itSupportsStylesApi(Progress, defaultProps, Object.keys(ProgressStylesApi), 'Progress');
+  itSupportsStylesApi(
```
Comment (rtivital) [inline src/mantine-core/src/components/Progress/Progress.test.tsx:26]: Please remove this test entirely

**Comment by rtivital** [comment]
Please also do this to merge this PR:
- Remove `showLabel` prop, it should detect label automatically based on `label` prop
- Rename `isIndeterminate` to `indeterminate` and `animate` to `animated`
- Add `indeterminate` and `animate` props to configurator
- Create new separate demo for new `label` prop and add it to documentation

File: src/mantine-core/src/components/Progress/Progress.test.tsx (hunk 1 above)
Comment (Ultra-Instinct-05) [reply src/mantine-core/src/components/Progress/Progress.test.tsx:26]: Are you saying to remove the `itSupportsStylesApi` test entirely for `Progress` (why ?) or just the changes _I made to it_ to pass the tests ? (I am still coming to terms on understating the test suite, but I am assuming this has something to do with the generated Mantine classes on components).

The rest of the below mentioned changes will be taken care.

**Comment by Ultra-Instinct-05** [comment]
Regarding `animate`, I feel it should remain `animate` (& not `animated`) to stay consistent with similar props elsewhere (Like the recently merged PR for `animate` for `Skeleton`), since both are booleans. It may confuse people later on that very similar props are named differently across components

File: src/mantine-core/src/components/Progress/Progress.test.tsx (hunk 1 above)
Comment (jerebtw) [reply src/mantine-core/src/components/Progress/Progress.test.tsx:26]: `itSupportsStylesApi` was completely removed see https://github.com/mantinedev/mantine/commit/4b4943af00e6ecbfbb49030abed6354f2989314d

Comment (Ultra-Instinct-05) [reply src/mantine-core/src/components/Progress/Progress.test.tsx:26]: Hmm.. Okay. Wasn't aware it was removed for all components.

**Comment by rtivital** [comment]
Thanks!

**Comment by jerebtw** [comment]
@Ultra-Instinct-05 could you please look in to the animation prop again

Because if you set the `size` to xl the progress animation looks buggy

yDyjY8LJBh

**Pull request merged**